File: FE/Seo-garden/투포인터_슬라이딩윈도우/Silver/2003_수들의합2.swift
let NM = readLine()!.split(separator: " ").map { Int($0)! }
let N = NM[0], M = NM[1]

let inputArr = readLine()!.split(separator: " ").map { Int($0)! }

var start = 0
var end = 0
var sum = 0
var count = 0

while end < N {
    if sum < M {
        sum += inputArr[end]        //end 포인트에 있는 값을 sum 넣고
        end += 1
    }
    while sum >= M {
        if sum == M {
            count += 1
        }
        sum -= inputArr[start]      //sum 이 M보다 크거나 같으면 start값만큼 빼고 start 포인트 +1
        start += 1
    }
}

print(count)

File: BE/tkdgml822/투 포인터/Silver/2003_수들의합2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class 2003_수들의합2 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[] arr = new int[N + 1];

        st = new StringTokenizer(br.readLine());
        int index = 0;
        while (st.hasMoreElements()) {
            arr[index++] = Integer.parseInt(st.nextToken());
        }

        int start_index = 0;
        int end_index = 0;
        int count = 0;
        int sum = 0;

        while (end_index <= N) {
            if (sum == M) {
                count++;
                sum += arr[end_index++];
            } else if (sum < M) {
                sum += arr[end_index++];
            } else {
                sum -= arr[start_index++];
            }
        }

        System.out.println(count);

    }
}
